packages/tools/devtools/devtools-browser-extension/src/popup/PopupScript.ts

```diff
@@ -19,7 +18,9 @@ import { browser } from "../Globals";
 //   If not, we may want to display an error message with a link to docs explaining how to
 //   use them.
 
-browser.tabs.query({ active: true, currentWindow: true }, (tab) => {
+const port = browser.runtime.connect({ name: "popup-background-channel" });
```

Comment:
Just to make sure I'm understanding correctly: this establishes a connection with the background script specifically? And the name we're providing is supposed to identify _this_ (the caller)?

Reply:
The name identifies the channel, more than the caller, IIRC. That's why I updated them at some point to be `<oneside>-<otherside>-<channel>`.

That said... I wonder if we need to use ports for this use case. [`browser.runtime.sendMessage()`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage) and [`browser.runtime.onMessage`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage) seem like they would do the trick, and I think they might be fine for communication between popup and background, which is more one-off than that between content and background. I _think_ the message sent from popup to background would only trigger listeners set up with `browser.runtime.onMessage()` in the background script, but if background script did `browser.runtime.sendMessage()` that would get to all pages of the extension (popup, options, content) so maybe we'd need to make sure that content ignores this one.

Or actually, if popup used `sendMessage()`, background could use `sendResponse()` on the message it gets to respond synchronously (an async option is available too, all documented in [onMessage()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage)) directly to popup, and I guess we don't need to have special considerations in content in that case.